### ISSUE TITLE
Accessing "this" in static methods

### ISSUE THREAD

**Issue by syranide** [google/closure-compiler#2956]
The following code emits a `Uncaught TypeError: this is not a constructor`. I've tried to decipher the different possible annotations but I can't really tell if there's an annotation for `create()` I need to add to make this work or if this pattern is simply not allowed (reading `this` in a static method).

Any input on this matter would be greatly appreciated, it's nigh impossible to search for any information on this due to lack of meaningful keywords. But I haven't been able to find anything on it.

It does NOT emit any warnings or errors when compiling this code.

PS. The pattern is rather stupid in this example, but it's meaningful in my context.

Compiled with `-O ADVANCED`.

----

```js
class X {
  print() {
    console.log("X");
  }

  static create() {
    let that = new (this);
    that.print();
  }
}

class Y extends X {
  print() {
    console.log("Y");
  }
}

Y.create();
```

It collapses to this, which is no longer valid:

```js
(function(){'use strict';
(function() {
  (new this).print();
})();
}).call(this)
```

**Comment by ChadKillingsworth** [comment]
Duplicate of #2397

**Comment by syranide** [comment]
My bad. Thank you!
